pkg/envoy/server.go

```diff
@@ -787,36 +787,18 @@ func (s *XDSServer) UpdateNetworkPolicy(ep logger.EndpointUpdater, policy *polic
 		policies = append(policies, networkPolicy)
 	}
 
-	if ep.HasSidecarProxy() { // Use sidecar proxy.
-		// If there are no L7 rules, we expect Envoy to NOT be configured with
-		// an L7 filter, in which case we'd never receive an ACK for the policy,
-		// and we'd wait forever.
-		// TODO: Remove this when we implement and inject an Envoy network filter
-		// into every Envoy listener to filter at L3/L4.
-		var hasL7Rules bool
-	Policies:
-		for _, p := range policies {
-			for _, pnp := range p.IngressPerPortPolicies {
-				for _, r := range pnp.Rules {
-					if r.L7 != nil {
-						hasL7Rules = true
-						break Policies
-					}
-				}
-			}
-			for _, pnp := range p.EgressPerPortPolicies {
-				for _, r := range pnp.Rules {
-					if r.L7 != nil {
-						hasL7Rules = true
-						break Policies
-					}
-				}
-			}
-		}
-		if !hasL7Rules {
-			wg = nil
+	nodeIDs := make([]string, 0, 1)
+	if ep.HasSidecarProxy() {
+		if ep.GetIPv4Address() == "" {
+			log.Fatal("Envoy: Sidecar proxy has no IPv4 address")
```

Comment:
Do we not support IPv6-only mode when running alongside Istio?

Reply:
So far Istio support has been IPv4 only, actually don't know if/when k8s and Istio support IPv6-only mode.